components/gfx/display_list/mod.rs

```diff
@@ -123,109 +177,105 @@ impl DisplayList {
         }
     }
 
+    fn text_index_stacking_context<'a>(&self,
+                                       stacking_context: &StackingContext,
+                                       clip_id: ClipId,
+                                       node: OpaqueNode,
+                                       traversal: &mut DisplayListTraversal<'a>,
+                                       point: &Point2D<Au>,
+                                       offset_lookup: &mut ScrollOffsetLookup,
+                                       result: &mut Vec<usize>) {
+        let mut point = *point - stacking_context.bounds.origin;
+        if stacking_context.scroll_policy == ScrollPolicy::Fixed {
+            let old_offset = offset_lookup.calculated_total_offsets.get(&clip_id).cloned();
+            offset_lookup.calculated_total_offsets.insert(clip_id, Point2D::zero());
```

Comment:
When can a clip_id correspond to different `StackingContext`? Perhaps it would be nice to describe how `Clip`s and `StackingContext` relate to each other, either here or in webrender's `CLIPPING.md`.

In any case, this looks correct to me.